src/main/java/org/swisspush/redisques/util/DequeueStatisticCollector.java

```diff
@@ -23,73 +23,85 @@ public DequeueStatisticCollector(Vertx vertx) {
         this.sharedData = vertx.sharedData();
     }
 
-    public void setDequeueStatistic(final String queueName, final DequeueStatistic dequeueStatistic) {
+    /**
+     *
+     * @param queueName
+     * @param dequeueStatistic
+     * @return an always completed future
+     */
+    public Future<Void> setDequeueStatistic(final String queueName, final DequeueStatistic dequeueStatistic) {
+        Promise<Void> promise = Promise.promise();
         sharedData.getLock(DEQUEUE_STATISTIC_LOCK_PREFIX.concat(queueName)).onComplete(lockAsyncResult -> {
             if (lockAsyncResult.failed()) {
                 log.error("Failed to lock dequeue statistic data for queue {}.", queueName, lockAsyncResult.cause());
```

Comment:
I tried the latest release of vertx-redisques (3.1.3) including the _DequeueStatisticsCollector_ class. However I got thousends of errors because the lock could not be made.

```
2024-04-25 09:10:01,463 vnecri2 int crush ERROR DequeueStatisticCollector - Failed to lock dequeue statistic data for queue tali-app-event-02378487.
io.vertx.core.VertxException: Timed out waiting to get lock dequeueStatisticLock.tali-app-event-02378487
at io.vertx.spi.cluster.hazelcast.HazelcastClusterManager.lambda$getLockWithTimeout$4(HazelcastClusterManager.java:235) ~[vertx-hazelcast-4.5.2.jar:4.5.2]
at io.vertx.core.impl.ContextImpl.lambda$executeBlocking$1(ContextImpl.java:190) ~[vertx-core-4.5.2.jar:4.5.2]
at io.vertx.core.impl.ContextInternal.dispatch(ContextInternal.java:276) ~[vertx-core-4.5.2.jar:4.5.2]
at io.vertx.core.impl.ContextImpl.lambda$internalExecuteBlocking$2(ContextImpl.java:209) ~[vertx-core-4.5.2.jar:4.5.2]
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[?:?]
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[?:?]
at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.106.Final.jar:4.1.106.Final]
at java.lang.Thread.run(Thread.java:829) ~[?:?]
```
One question: Why do we use sharedData for this? In all the features implemented in gateleen, vertx-rest-storage and vertx-redisques we always avoided using this. In gateleen and vertx-rest-storage we even have redis based implementations for Locks. See https://github.com/swisspost/gateleen/blob/0b83c698bf5f59d3b0dcf3686caedb27f112cceb/gateleen-core/src/main/java/org/swisspush/gateleen/core/lock/impl/RedisBasedLock.java#L28

I would suggest to use something like this and remove the sharedData stuff

Reply:
I would hope that it performs better compared to keeping the stats in redis. I mean it is the standard mechanism to share data across Verticles and does work in a cluster environment. We don't need to be the data to be persistent, so we don't have the need to put it into redis really. In our project we use redisques in a vertx cluster, so sharedData comes into play anyway.

But I wasn't aware that we intentionally tried to avoid sharedData so far and what the reasons are, happy to discuss.